store/postgres/src/primary.rs

```diff
@@ -322,37 +410,71 @@ impl<'a> Connection<'a> {
         }
     }
 
+    /// Signal any copy process that might be copying into one of these
+    /// deployments that it should stop. Copying is cancelled whenever we
+    /// remove the assignment for a deployment
+    fn cancel_copies(&self, ids: Vec<DeploymentId>) -> Result<(), StoreError> {
+        use active_copies as ac;
+
+        update(ac::table.filter(ac::dst.eq_any(ids)))
+            .set(ac::cancelled_at.eq(sql("now()")))
+            .execute(self.0.as_ref())?;
+        Ok(())
+    }
+
     /// Delete all assignments for deployments that are neither the current nor the
     /// pending version of a subgraph and return the deployment id's
     fn remove_unused_assignments(&self) -> Result<Vec<EntityChange>, StoreError> {
-        const QUERY: &str = "
-    delete from subgraphs.subgraph_deployment_assignment a
-    where not exists (select 1
-                        from subgraphs.subgraph s, subgraphs.subgraph_version v
-                       where v.id in (s.current_version, s.pending_version)
-                         and v.deployment = a.id)
-    returning a.id
-    ";
-        #[derive(QueryableByName)]
-        struct Removed {
-            #[sql_type = "Text"]
-            id: String,
-        }
+        use deployment_schemas as ds;
+        use subgraph as s;
+        use subgraph_deployment_assignment as a;
+        use subgraph_version as v;
 
-        Ok(diesel::sql_query(QUERY)
-            .load::<Removed>(self.0.as_ref())?
+        let active = v::table
```

Comment:
`active` has a different meaning now, so to not be ambiguous this could be called `in_use` or something.

Reply:
Good point. Changing this to `named`